cfme/middleware/datasource.py

```diff
@@ -6,6 +8,8 @@
 from cfme.web_ui.menu import nav, toolbar as tb
 from mgmtsystem.hawkular import CanonicalPath
 from utils import attributize_string
+from utils.appliance import Navigatable, CurrentAppliance
```

Comment:
I believe you can remove CurrentAppliance import and setting the appliance attribute for all of these classes.